### PR TITLE
Azure tests

### PR DESCRIPTION
**Pull request by Taluu**
Unit tests and docs on azure, by @phpetra (superseeds #1)

### CODE REVIEW

**Comment by Taluu** [comment]
I rebased and reworked a little bit what you did @phpetra, does it look good to you ? If so, I will merge it, so that @Soullivaneuh can review the whole thing

File: OAuth/ResourceOwner/AzureResourceOwner.php [hunk 1]
```diff
@@ -11,10 +11,11 @@
 
 namespace HWI\Bundle\OAuthBundle\OAuth\ResourceOwner;
 
+use InvalidArgumentException;
```
Comment (soullivaneuh) [inline OAuth/ResourceOwner/AzureResourceOwner.php]: Do not import base PHP classes. Use `new \InvalidArgumentException();` instead.

Comment (Taluu) [inline OAuth/ResourceOwner/AzureResourceOwner.php]: fix'd

**Comment by soullivaneuh** [comment]
:+1: Waiting merge for a full review.

**Comment by Taluu** [comment]
I'll wait until the full review is done before squashing things (even though I'm not a big adept of squashing everything. but meh)

**Comment by phpetra** [comment]
@Taluu It certainly looks good to me! 
Please go ahead and squash, or whatever it is ;-)

@Soullivaneuh /cc

**Pull request merged**